src/analyses/goto_check.cpp

```diff
@@ -1898,36 +1920,62 @@ void goto_checkt::goto_check(
     current_target = it;
     goto_programt::instructiont &i=*it;
 
-    flag_resett flag_resetter;
+    flag_resett resetter;
```

Comment:
Given that every call requires the instruction to be passed (and the instruction is always the same), should the constructor already take that?

Reply:
fixed